src/livePreview/eventManager/postMessageEvent.hooks.ts

```diff
@@ -52,10 +54,43 @@ export function useOnEntryUpdatePostMessageEvent(): void {
             if (!ssr) {
                 onChange();
             }
+            if (VisualBuilder.VisualBuilderGlobalState.value.audienceMode) {
+                recalculateVariantClasses();
+            }
         }
     );
 }
 
+function recalculateVariantClasses(): void {
+    const elements = document.querySelectorAll("[data-cslp]");
+    const observer = new MutationObserver((mutations, obs) => {
+        mutations.forEach((mutation) => {
+            if (
+                mutation.type === "attributes" &&
+                mutation.attributeName === "data-cslp"
+            ) {
+                const element = mutation.target as HTMLElement;
+                const dataCslp = element.getAttribute("data-cslp");
+                if (!dataCslp) return;
+                if (
+                    dataCslp.startsWith("v2:") &&
+                    element.classList.contains("visual-builder__base-field")
+                ) {
+                    element.classList.remove(
+                        visualBuilderStyles()["visual-builder__base-field"],
+                        "visual-builder__base-field"
+                    );
+                }
+                obs.disconnect();
+            }
+        });
+    });
+
+    elements.forEach((element) => {
+        observer.observe(element, { attributes: true });
+    });
+}
+
```

Comment:
I am not sure about adding mutation observe for each element. 
@faraazb @csAyushDubey Can you please check this?

Reply:
yup. I also feel this isn't right approach. tried alternatives that didn't work. I'll still check more on this